numpy/lib/financial.py

```diff
@@ -139,6 +151,8 @@ def fv(rate, nper, pmt, pv, when='end'):
 
 
 def _pmt_dispatcher(rate, nper, pv, fv=None, when=None):
+    warnings.warn('pmt is deprecated and will be removed from NumPy 1.20.',
+                   DeprecationWarning, stacklevel=3)
     return (rate, nper, pv, fv)
```

Comment:
@shoyer: Is the dispatcher the right place for this? What if `array_function` is disabled via the env var (if that still exists)?

Reply:
FWIW: This is how `asscalar` is deprecated: https://github.com/numpy/numpy/blob/1185880f153c5bbc2081a17121617beb588cfb1f/numpy/lib/type_check.py#L551

Reply:
That seems to answer the question and looks fine. @shoyer did that `asscalar` deprecation in the same way says git blame.

> What if `array_function` is disabled via the env var (if that still exists)?

That only still exists as an escape hatch in case of unexpected issues with the mechanism, it shouldn't be used otherwise. No need to worry about that I'd say, since even if it is used the impact is very minor.